### PR TITLE
Add make_llvm_arm tool support

### LINKED ISSUE #503: Support for "LLVM Embedded Toolchain for Arm"
[LLVM Embedded Toolchain for Arm](https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm) is a new LLVM-based toolchain, I have been considering adding support for. Maybe one question is how do we call it? `clang_arm` is a little bit confusing since `armclang` exists. `llvm_arm`?

### PR DESCRIPTION
**Pull request by mbrossard**
Add support for [LLVM Embedded Toolchain for Arm](https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm). Fixes #503.

### CODE REVIEW

File: project_generator/tools/llvm_arm.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+import logging
+
+from .makefile import MakefileTool
+
+logger = logging.getLogger('progen.tools.gccarm')
```
Comment (0xc0170) [inline project_generator/tools/llvm_arm.py:20]: the logger should be `llvm_arm`

**Review by 0xc0170 — COMMENTED** [review]
Let's rebase and it should be good to go (just one nit there)

File: project_generator/tools/llvm_arm.py (hunk 1 above)
Comment (mathias-arm) [reply project_generator/tools/llvm_arm.py]: Fixed

**Review by 0xc0170 — APPROVED** [review]

**Comment by 0xc0170** [comment]
v0.12.0 released

**Pull request merged**